csharp/private/rules/resgen.bzl

```diff
@@ -0,0 +1,50 @@
+# Label of the template file to use.
+_TEMPLATE = "@d2l_rules_csharp//csharp/private:rules/Template.csproj"
+_TEMPLATE_EMBEDDED_RESOURCE = "        <EmbeddedResource Include=\"%s\" />"
+
+def _csproj_embedded_resource(resx_files):
+    result = ""
+    for src in resx_files:
+        result += _TEMPLATE_EMBEDDED_RESOURCE % (src.basename)
+    return result
+
+def _csharp_resx_impl(ctx):
+    proj_name = ctx.file.csproj.basename[:-(len(ctx.file.csproj.extension)+1)]
+
+    csproj_output = ctx.actions.declare_file(ctx.file.csproj.basename)
+    embedded_resources = _csproj_embedded_resource(ctx.files.srcs)
+    ctx.actions.expand_template(
+        template = ctx.file._template,
+        output = csproj_output,
+        substitutions = {
+            "{FRAMEWORK}": ctx.attr.target_frameworks[0],
+            "{RESOURCES}": embedded_resources,
+        },
+    )
+
+    files = depset(direct = csproj_output)
+    runfiles = ctx.runfiles(files = csproj_output)
+    return [DefaultInfo(files = files, runfiles = runfiles)]
+
+csharp_resx = rule(
+    implementation = _csharp_resx_impl,
+    attrs = {
+        "srcs": attr.label_list(
+            mandatory = True, 
+            allow_files = True
+        ),
+        "csproj": attr.label(
+            mandatory = True, 
+            allow_single_file = True
+        ),
+        "target_frameworks": attr.string_list(

```

Comment:
I don't like having to specify the target framework for this, but I noticed when working with localized resources (e.g. `Strings.resx`, `Strings.en-FR.resx`) that dlls were compiled for each localization. I have not had time yet to thoroughly investigate the impact of different framework targets on the compiled CLR binary `.resources`, or the output `.dll` per culture.

For example, for `hello.csproj`, there would be a compiled resources dll in `bin/en-FR/hello.resources.dll`.